packages/mui-material/src/OutlinedInput/OutlinedInput.js

```diff
@@ -124,13 +126,29 @@ const OutlinedInput = React.forwardRef(function OutlinedInput(inProps, ref) {
 
   const classes = useUtilityClasses(props);
 
+  const muiFormControl = useFormControl();
+  const fcs = formControlState({
+    props,
+    muiFormControl,
+    states: ['required'],
+  });
+
   return (
     <InputBase
       components={{ Root: OutlinedInputRoot, Input: OutlinedInputInput, ...components }}
       renderSuffix={(state) => (
         <NotchedOutlineRoot
           className={classes.notchedOutline}
-          label={label}
+          label={
+            label && fcs.required ? (
+              <React.Fragment>
+                {label}
+                {'\u00a0*'}
```

Comment:
I'm not sure about this. In FormLabel, we use `&thinsp;`. Before #15132, we were using \u2009.

Reply:
I think it should be non-breaking, as a dangling asterisk would not look good. If you're concerned with its width, we could use &#8239; (narrow non-breaking space)

Reply:
Hi! It seems html entities are more prevalent in the source code. Although `\u2009` and `\u00a0` are still using in `FormLabel.test.js` `TextField.js`, `TextField.test.js` components. We can also change them to `&thinsp;` and `&nbsp;`. If I get it right and you were talking about consistency in the source code.

Reply:
I replaced it with non-breaking space. Here's a codesandbox with the current build: https://codesandbox.io/s/agitated-germain-o11s7?file=/src/Demo.tsx